[charts] Add zoom slider (#17496)
mui/mui-x: docs/data/charts/zoom-and-pan/zoom-and-pan.md

{{"demo": "ZoomFilterMode.js"}}

## Zoom slider 🧪

:::warning
This feature is unstable and its API, visuals and/or behavior may change in future minor or patch releases.
:::

You can provide an overview and allow the manipulation of the zoomed area by setting the `zoom.slider.enabled` property on the axis config.

{{"demo": "ZoomSlider.js"}}

> Optionally, you can set the `zoom.slider.size` property to customize the zoom slider's size, i.e., the height on an x-axis and the width on a y-axis.

Reviewer: ```suggestion
Optionally, you can set the `zoom.slider.size` property to customize the zoom slider's size, that is, the height on an x-axis and the width on a y-axis.
```
